mesonbuild/backend/ninjabackend.py

```diff
@@ -502,13 +502,12 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
         # hence we disable them if 'cuda' is enabled globally. See also
         # - https://github.com/mesonbuild/meson/pull/9453
         # - https://github.com/mesonbuild/meson/issues/9479#issuecomment-953485040
-        self._allow_thin_archives = PerMachine[bool](
-            'cuda' not in self.environment.coredata.compilers.build,
-            'cuda' not in self.environment.coredata.compilers.host) if self.environment else PerMachine[bool](True, True)
-        if not self._allow_thin_archives.build:
-            mlog.debug('cuda enabled globally, disabling thin archives for build machine, since nvcc/nvlink cannot handle thin archives natively')
-        if not self._allow_thin_archives.host:
-            mlog.debug('cuda enabled globally, disabling thin archives for host machine, since nvcc/nvlink cannot handle thin archives natively')
+        self.allow_thin_archives = PerMachine[bool](True, True)
+        if self.environment:
+            for for_machine in (MachineChoice.BUILD, MachineChoice.HOST):
+                if 'cuda' in self.environment.coredata.compilers[for_machine]:
+                    mlog.debug('cuda enabled globally, disabling thin archives for {} machine, since nvcc/nvlink cannot handle thin archives natively'.format(for_machine))
```

Comment:
for_machine prints as "build machine" etc. so using `{} machine` is redundant I think.